### PR TITLE
Run spec test for classic/fast-interp in NuttX CI

### PR DESCRIPTION
**Pull request by no1wudi**
Notice: It will take a long time to run, up to 4 hour, see: https://github.com/no1wudi/wasm-micro-runtime/actions/runs/6966974742 and https://github.com/no1wudi/wasm-micro-runtime/actions/runs/6969744155

### CODE REVIEW

File: .github/workflows/spec_test_on_nuttx.yml [hunk 1]
```diff
@@ -107,8 +120,11 @@ jobs:
...
         run: |
-          find nuttx/boards -name defconfig | xargs sed -i '$a\CONFIG_INTERPRETERS_WAMR=y\nCONFIG_INTERPRETERS_WAMR_STACKSIZE=32768\nCONFIG_INTERPRETERS_WAMR_AOT=y\nCONFIG_INTERPRETERS_WAMR_FAST=y\nCONFIG_INTERPRETERS_WAMR_LOG=y\nCONFIG_INTERPRETERS_WAMR_LIBC_BUILTIN=y\nCONFIG_INTERPRETERS_WAMR_REF_TYPES=y\nCONFIG_INTERPRETERS_WAMR_ENABLE_SPEC_TEST=y\nCONFIG_INTERPRETERS_WAMR_SHARED_MEMORY=y\nCONFIG_INTERPRETERS_WAMR_BULK_MEMORY=y\n'
-          find nuttx/boards -name defconfig | xargs sed -i '$a\CONFIG_EOL_IS_LF=y\nCONFIG_ARM_SEMIHOSTING_HOSTFS=y\nCONFIG_ARM_SEMIHOSTING_HOSTFS_CACHE_COHERENCE=y\nCONFIG_RISCV_SEMIHOSTING_HOSTFS=y\nCONFIG_FS_HOSTFS=y\nCONFIG_LIBC_FLOATINGPOINT=y\n'
+          find nuttx/boards -name defconfig | xargs sed -i '$a\${{ env.WAMR_COMMON_OPTION }}'
+      
+      - name: Enable WAMR Interpreter for NuttX
+        run: |
+          find nuttx/boards -name defconfig | xargs sed -i '$a\${{ matrix.wamr_running_mode_option }}'
 
       - name: Enable additional features for NuttX on RI5C-V
```
Comment (wenyongh) [inline .github/workflows/spec_test_on_nuttx.yml:129]: Found a typo here, could you help change `RI5C-V` to `RISC-V`?

Comment (no1wudi) [reply .github/workflows/spec_test_on_nuttx.yml]: Done

**Comment by wenyongh** [comment]
Seems that riscv-64 and aarch64 are not tested, will we enable them in the future?

**Comment by no1wudi** [comment]
> Seems that riscv-64 and aarch64 are not tested, will we enable them in the future?

I'll enable them soon ( include gc spec test).

File: tests/wamr-test-suites/test_wamr.sh [hunk 2]
```diff
@@ -15,7 +15,7 @@ function help()
     echo "test_wamr.sh [options]"
     echo "-c clean previous test results, not start test"
     echo "-s {suite_name} test only one suite (spec|wasi_certification|wamr_compiler)"
-    echo "-m set compile target of iwasm(x86_64|x86_32|armv7_vfp|thumbv7_vfp|riscv64_lp64d|riscv64_lp64|aarch64)"
+    echo "-m set compile target of iwasm(x86_64|x86_32|armv7|armv7_vfp|thumbv7|thumbv7_vfp|riscv32|riscv32_ilp32f|riscv32_ilp32d|riscv64|riscv64_lp64f|riscv64_lp64d|aarch64|aarch64_vfp)"
```
Comment (wenyongh) [inline tests/wamr-test-suites/test_wamr.sh:18]: Could we output the message in multiple lines?

File: tests/wamr-test-suites/test_wamr.sh [hunk 3]
```diff
@@ -65,6 +65,7 @@ ENABLE_QEMU=0
 QEMU_FIRMWARE=""
 # prod/testsuite-all branch
 WASI_TESTSUITE_COMMIT="ee807fc551978490bf1c277059aabfa1e589a6c2"
+TARGET_LIST=("AARCH64" "AARCH64_VFP" "ARMV7" "ARMV7_VFP" "THUMBV7" "THUMBV7_VFP" "RISCV32" "RISCV32_ILP32F" "RISCV32_ILP32D" "RISCV64" "RISCV64_LP64F" "RISCV64_LP64D")
```
Comment (wenyongh) [inline tests/wamr-test-suites/test_wamr.sh:68]: Same as above.

File: tests/wamr-test-suites/test_wamr.sh (hunk 2 above)
Comment (no1wudi) [reply tests/wamr-test-suites/test_wamr.sh]: Please review again

File: tests/wamr-test-suites/test_wamr.sh [hunk 4]
```diff
@@ -15,7 +15,8 @@ function help()
     echo "test_wamr.sh [options]"
     echo "-c clean previous test results, not start test"
     echo "-s {suite_name} test only one suite (spec|wasi_certification|wamr_compiler)"
-    echo "-m set compile target of iwasm(x86_64|x86_32|armv7_vfp|thumbv7_vfp|riscv64_lp64d|riscv64_lp64|aarch64)"
+    echo "-m set compile target of iwasm(x86_64|x86_32|armv7|armv7_vfp|thumbv7|thumbv7_vfp|\
+riscv32|riscv32_ilp32f|riscv32_ilp32d|riscv64|riscv64_lp64f|riscv64_lp64d|aarch64|aarch64_vfp)"
```
Comment (wenyongh) [inline tests/wamr-test-suites/test_wamr.sh:19]: The code looks not so friendly, how about:
```bash
    echo "-m set compile target of iwasm(x86_64|x86_32|armv7|armv7_vfp|thumbv7|thumbv7_vfp|"
    echo "                               riscv32|riscv32_ilp32f|riscv32_ilp32d|riscv64|"
    echo "                               riscv64_lp64f|riscv64_lp64d|aarch64|aarch64_vfp)"
```

Comment (no1wudi) [reply tests/wamr-test-suites/test_wamr.sh:19]: OK

**Comment by wenyongh** [comment]
> @wenyongh Spectest workflow is ready to cover more target by add board and target configuration now, but need further debug to pass all test on the target that commented out, e.g. https://github.com/no1wudi/wasm-micro-runtime/actions/runs/7000142793
> 
> Please review this change, if it's OK for you we can consider merge it first and then fix the error to enable all targets step by step.

@no1wudi I am OK for the changes, my concern is that the CIs run failed, could we disable the failed targets, merge the PR, and then enable them one by one?

**Pull request merged**